cluster-api/examples/manifests/machinesets.yaml

```diff
@@ -0,0 +1,24 @@
+apiVersion: "cluster.k8s.io/v1alpha1"
+kind: MachineSet
+metadata:
+  name: my-first-machineset
+spec:
+  replicas: 3
+  template:
+    spec:
+      providerConfig: >
+        {
+          "apiVersion": "gceproviderconfig/v1alpha1",
+          "kind": "GCEProviderConfig",
```

Comment:
I like that you didn't add the controller to cluster-api-gcp (since we need to move the common code out of it), but shouldn't the yaml specific to GCP be in cluster-api-gcp/cloud/google and equivalent ext-apiserver dir?

Do you plan to have this installed as part of the cluster provisioning as a next step?